### ISSUE TITLE
MemoryReactor incompletely implements IReactorSocket

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6349]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6349|
|Type|defect|
|Created|2013-03-03 18:52:10Z|
|Branch|https://github.com/twisted/twisted/tree/memoryreactor-adoptconn-6349|

It has no `adoptStreamConnection` method.

<details><summary>Searchable metadata</summary>

```
trac-id__6349 6349
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_memoryreactor_adoptconn_6349 branches/memoryreactor-adoptconn-6349
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1362336730000000 1362336730000000
changetime__1362428760000000 1362428760000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [37374]) Branching to 'memoryreactor-adoptconn-6349'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/memoryreactor-adoptconn-6349)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_3"></a>|@tomprince set owner to @exarkun|
|-|-|

Looks good. I'd consider creating a test module for `proto_helpers` and putting the verification in a test case there. (This is perhaps overkill for this test, but I think we want a verified test reactor, at some point, so creating a module to put tests in would be a good start).

Please merge, after optionally creating the test module.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Thanks.  I agree some testing would be better, but I don't want to work on that at the moment.  `proto_helpers` isn't the ultimate location for any solid testing tools anyway, since the right location will not be inside a `test` package.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [37386]) Merge memoryreactor-adoptconn-6349

Author: exarkun
Reviewer: tom.prince
Fixes: [#6349](https://github.com/twisted/twisted/issues/6349)

Finish the implementation of `IReactorSocket` by `MemoryReactor` by adding an
`adoptStreamConnection` method.
